### PR TITLE
Add github action to release helm charts

### PR DESCRIPTION
**Pull request by yonatankahana**
Add github action to create chart releases automatically once the chart version is bumped.
After this PR and creation of orphan branch named `gh-pages` - this repository could be used in `helm repo add` command.

```console
helm repo add nfs-subdir-external-provisioner https://kubernetes-sigs.github.io/nfs-subdir-external-provisioner/
```

For example, my fork can be used:

```console
$ helm repo add nfs-subdir-external-provisioner https://yonatankahana.github.io/nfs-subdir-external-provisioner/
"nfs-subdir-external-provisioner" has been added to your repositories
$ helm search repo
NAME                                           	CHART VERSION	APP VERSION	DESCRIPTION                                       
nfs-subdir-external-provisioner/nfs-subdir-external-provisioner	4.0.0        	3.1.0      	nfs-subdir-external-provisioner is an automatic...
```

To create the `gh-pages` branch use: `git checkout --orphan gh-pages`.

### CODE REVIEW

**Comment by wongma7** [comment]
I'm not sure if I have permission to create a new branch, will try.

Also will need make sure the gh-pages branch gets marked unprotected so that the action has permission to write to it (obviously github action cannot sign a cla):

https://github.com/kubernetes/test-infra/pull/20323/files

File: .github/workflows/helm-chart-release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
```
Comment (wongma7) [inline .github/workflows/helm-chart-release.yml:22]: this part unnecessary, chart releaser action should be enough like at https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/.github/workflows/helm-chart-release.yaml#L22

Comment (yonatankahana) [reply .github/workflows/helm-chart-release.yml]: excellent, i deleted that.

**Comment by yonatankahana** [comment]
> Also will need make sure the gh-pages branch gets marked unprotected so that the action has permission to write to it (obviously github action cannot sign a cla)

follow: https://github.com/kubernetes/test-infra/pull/20453

**Comment by kmova** [comment]
Was able to push in the gh-pages branch. @yonatankahana @wongma7 - please check if you would like to modify/update the default index.md with any further information.

**Comment by yonatankahana** [comment]
> please check if you would like to modify/update the default index.md with any further information.

i have left 2 cosmetics comments

File: .github/workflows/helm-chart-release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.1.0
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (kmova) [inline .github/workflows/helm-chart-release.yml:25]: @yonatankahana - could you raise an issue with the gh-admin to setup GITHUB_TOKEN to this repo. Probably something that comes from the organization's secrets. (similar to the PR that you raised to resolve the branch protection rules)

Comment (kmova) [reply .github/workflows/helm-chart-release.yml:25]: Or, are we good with the default permissions availble on GITHUB_TOKEN enabled for github actions. https://docs.github.com/en/actions/reference/authentication-in-a-workflow

Comment (yonatankahana) [reply .github/workflows/helm-chart-release.yml:25]: hope that i understand you correctly, do you mean that the token will be k8s bot token?
if so, this may be related? https://github.com/kubernetes/test-infra/pull/20291, https://github.com/kubernetes/test-infra/pull/20425

Comment (kmova) [reply .github/workflows/helm-chart-release.yml:25]: yes. on the lines of the above. Let me go ahead with this merge and see if we are good based on the discussions in the above PRs.

Comment (yonatankahana) [reply .github/workflows/helm-chart-release.yml:25]: for example, they use the default token here: https://github.com/kubernetes/kube-state-metrics/releases

**Comment by kmova** [comment]
@yonatankahana - just wondering if we should keep the charts in the following location => `deploy/helm/charts`, to have all the different ways of deployments in the `deploy` folder. 

The chart release can be configured with a custom directory as follows: https://github.com/openebs/node-disk-manager/blob/master/.github/workflows/release-charts.yml#L33

**Comment by yonatankahana** [comment]
> @yonatankahana - just wondering if we should keep the charts in the following location => `deploy/helm/charts`, to have all the different ways of deployments in the `deploy` folder.
> 
> The chart release can be configured with a custom directory as follows: https://github.com/openebs/node-disk-manager/blob/master/.github/workflows/release-charts.yml#L33

I always prefer to stick with as many defaults as possible, it leads to conventions. If everyone uses the default it will be easier for the user to find what they are looking for because it will be the same in every (git as helm) repository.
In a short search I did in other repositories I see that there is a lack of convention and everyone places the chart in a different place so that in any case - there is no real convention other than this default.
(One way or another - the `deploy/` folder with the `deploy/objects/` feels a bit like duplication.)

tl;dr
I'll go with the default: `charts/`.

**Comment by kmova** [comment]
@yonatankahana,  I agree with: 
> stick with as many defaults as possible

**Review by kmova — APPROVED** [review]
/lgtm
/approve
